external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutMessageId.java

```diff
@@ -18,36 +18,88 @@
 
 package org.apache.storm.kafka.spout;
 
+import java.io.IOException;
```

Comment:
I think maybe we should split this object. We don't need half of the fields for metadata, and several fields aren't needed for emit/ack handling either. Maybe have an object with thread name and topology id for metadata, and have KafkaSpoutMessageId only contain the data we need for ack handling? That way we can also avoid doing stuff like storing the topic partition and offset in metadata, which IMO makes no sense.

Reply:
Done